DEVELOPER_STARTING_GUIDE.md

```diff
@@ -0,0 +1,190 @@
+
+### Storybook Helpers
+
+To make things easier, @mariohamann has authored some [Storybook Helpers](https://socket.dev/npm/package/@mariohamann/wc-storybook-helpers)
+
+These will allow you to do things like present various properties along two axes for quick visual comparison.
+![image](https://github.com/solid-design-system/solid/assets/39494579/b6b4c3c4-47b0-4497-a1f6-1778e3109c03)
+
+## Work Process
+
+### Components
+
+If you are working on a component, follow the steps below:
+
+1. Move the component folder from `packages/components/src/_components` to `packages/components/src/components` or run `$pnpm plop` to generate a blank component template.
+2. **Open the component's Storybook Docs or default Story.** Ensure this is working properly. You'll want to have a place to observe the component as you work.
+3. **Check the [Shoelace repo components folder](https://github.com/shoelace-style/shoelace/tree/next/src/components) for the latest published component code.** If you need to re-copy updated source code, ensure you _copy the whole component folder not just the component file_ (includes tests and styles) AND rename contained imports, variables, components etc... prefixed with `sl` to `sd` (e.g. `sl-component` becomes `sd-component` ) and likewise `Shoelace` to `Solid` (e.g. `ShoelaceFormControl` becomes `SolidFormControl`).
+4. **Remove any properties from the copied Shoelace code that are irrelevant to our design** (e.g. `pill` is a re-occurring prop in Shoelace that alters a component's border radius, our designs do not offer this). Refer to the [Figma design documentation](https://www.figma.com/files/1075429990769806468/project/67503549/Solid-DS-Documentation?fuid=883643809929820461) for the component you are working on.
+5. **Compare the old UI component to the newly adapted Shoelace component.** Refer to the [UI Component Library](https://component-library.dev.fe.union-investment.de/integration/storybook/index.html) to get an idea of what is being changed. This helps to see what is being lost or improved upon and should be summarized in the migration guide.
+6. **Keep the technical props and features of the Shoelace component that feel relevant.** It is easier to decide which ones can be removed at the end.
+7. **Convert the component styles to use a Tailwind first approach.** Shoelace does not use Tailwind and includes CSS files for every component. These should be carefully inspected and adapted. Work your way through the render() method of the component and remove CSS classes one by one, replacing them with a combination of Tailwind classes that implement our design. Keep in mind our designs may differ from the original styles applied by Shoelace, however, there are many small details (regarding `display`, `position`, or interaction states etc.) that should be carefully observed and adapted.
```

Comment:
Maybe we should also mentioned here that our rule is 'mobile first'.  Developers should keep in mind to use TailwindCSS accordingly.

Reply:
Is this really still the case? isn't the device no longer a criteria for us as we don't want to care about where our components are used?

Reply:
That's correct, we don't prioritize how and where the Solid components are integrated and for which use case they will ultimately be used. However, we still build our components following the Mobile First approach. For instance, "text-3xl lg:text-4xl." We need to do this to ensure that our components are displayed correctly on both desktop and mobile. In essence, we initially oriented ourselves this way, and it's essential to stay true to the principle to avoid mixed desktop/mobile-first approaches everywhere.

Reply:
added a suggestion below @yoezlem. What do you think?

Reply:
LGTM